### PR TITLE
Update telemetry format

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes

- [x] Update the `npm version` hook to generate the dist files under the tag name
- [x] Use a gulp plugin to include this SDK's package version into the telemetry object. 
- [x] Update the telemetry format sent on the requests.
- [x] Update the README suggesting the use of a newer Lock and Auth0JS version. This is also set on the bower.json file.
- [x] Update lock and auth0js version to latest (Fixes telemetry extensibility)
- [x] Release 3.0.2

Telemetry format will end up looking like this for Lock requests:
```
{"name":"angular-lock","version":"3.0.2","env":{"lock.js":"11.16.2","auth0.js":"9.10.4"}}
```

### Testing

Inspect the request headers using the web tools. Look for the auth0-client header and b64 decode it.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -3,12 +3,17 @@ var gulp = require('gulp'),
...
+    injectVersion = require('gulp-inject-version'),
     sourceFiles = [
       'angular-lock.js'
     ];
 
 gulp.task('build', function() {
   gulp.src(sourceFiles)
+    .pipe(injectVersion({
+      prepend: '',
+      replace: /%%PACKAGE_VERSION%%/g
```
Comment (damieng) [inline gulpfile.js]: Where does the replacement version number come from?

Comment (lbalmaceda) [reply gulpfile.js]: I'm using https://www.npmjs.com/package/gulp-inject-version. Takes the `version` property from the `package.json` file in the root and replaces it globally wherever the placeholder is present.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,3122 @@
+{
+  "name": "angular-lock",
```
Comment (luisrudge) [inline package-lock.json]: all repos use yarn, so let's remove this file. if you want to add a lock file, use yarn.

Comment (lbalmaceda) [reply package-lock.json]: Done

File: bower.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "main": "angular-lock.js",
```
Comment (luisrudge) [inline bower.json]: You're missing a version update in this file as well

Comment (lbalmaceda) [reply bower.json]: Removed that field as per this comment https://github.com/bower/bower/issues/2389#issuecomment-259408378.

**Pull request merged**